.github/workflows/ci.yaml

```diff
@@ -0,0 +1,192 @@
+name: OHCL Build and Validation
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    tags:
+      - '**'
+
+jobs:
+  build_linux:
+    name: Build Linux kernel
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: ['x64', 'cvm-x64', 'arm64']
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential libncurses-dev bison flex libssl-dev libelf-dev
+          [ "${{ matrix.arch }}" == "arm64" ] && sudo apt-get install -y gcc-aarch64-linux-gnu || true
```

Comment:
`|| true` is there to prevent a pipeline failure when `matrix.arch != "arm64"`, this piece of code is pretty much equivalent to

```
if [ "${{ matrix.arch }}" == "arm64" ]; then
    sudo apt-get install -y gcc-aarch64-linux-gnu
fi
```